src/sccz80/codegen.c

```diff
@@ -5192,14 +5192,9 @@ void constbc(int32_t val)
 
 void addbchl(int val)
 {
-    if ( c_cpu == CPU_Z80N ) {
-        ot("add\thl,");
-        outdec(val); nl();
-    } else {
-        ot("ld\tbc,");
-        outdec(val);
-        outstr("\n\tadd\thl,bc\n");
-    }
+    ot("ld\tbc,");
+    outdec(val);
+    outstr("\n\tadd\thl,bc\n");
```

Comment:
and why not using `add hl,nnnn` with z80n? I mean, it feels correct change to me, because the function has name `addbchl`, but otherwise `add hl,nnnn` is more optimal on Z80N if resulting BC is not important.

Reply:
I found it was interfering with some of the other rules.

The plan was to do a replacement in z80n_rules.1 but that needs some thought since a long + constant addition usually looks something like this:

```
ld bc,nnnn
add hl,bc
ex de,hl
ld bc,nnnn
adc hl,bc
ex de, hl
```
 
Which needs the carry flag from the `add hl,bc`

Reply:
then there's no replacement, `add hl,nnnn` preserves carry flag and any checks for overflow will be more costly than regular Z80 `add hl,r16`.

(btw the other z80n extension `add hl,a` destroys carry flag with meaningless value - probably result of typo in VHDL, unfinished real carry reserving 17 bits and writing carry from b16, but letting adder to set only 16 of them, so the b16 is "whatever" fpga has there in the temp).